File: webflux-dialogue/src/main/java/com/study/webflux/rag/domain/credit/model/CreditTransactionId.java
package com.study.webflux.rag.domain.credit.model;

import java.util.UUID;

public record CreditTransactionId(
	String value
) {
	public CreditTransactionId {
		if (value == null || value.isBlank()) {
			throw new IllegalArgumentException("creditTransactionId cannot be null or blank");
		}
	}

	public static CreditTransactionId of(String value) {
		return new CreditTransactionId(value);
	}

	public static CreditTransactionId generate() {
		return new CreditTransactionId(UUID.randomUUID().toString());
	}
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/credit/model/CreditTransactionId.kt
package com.miyou.app.domain.credit.model

import java.util.UUID

data class CreditTransactionId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "creditTransactionId cannot be null or blank" }
    }

    companion object {
        @JvmStatic
        fun of(value: String): CreditTransactionId = CreditTransactionId(value)

        @JvmStatic
        fun generate(): CreditTransactionId = CreditTransactionId(UUID.randomUUID().toString())
    }

    fun value(): String = value
}
